modules/elastic/src/main/scala/ch/epfl/bluebrain/nexus/kg/query/BaseElasticQueries.scala

```diff
@@ -0,0 +1,165 @@
+package ch.epfl.bluebrain.nexus.kg.query
+
+import cats.instances.string._
+import ch.epfl.bluebrain.nexus.commons.es.client.ElasticClient
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient
+import ch.epfl.bluebrain.nexus.commons.types.search.{Pagination, QueryResults, Sort, SortList}
+import ch.epfl.bluebrain.nexus.kg.core.{ConfiguredQualifier, Qualifier}
+import ch.epfl.bluebrain.nexus.kg.core.domains.DomainId
+import ch.epfl.bluebrain.nexus.kg.core.IndexingVocab.PrefixMapping.rdfTypeKey
+import ch.epfl.bluebrain.nexus.kg.core.Qualifier._
+import ch.epfl.bluebrain.nexus.kg.core.organizations.OrgId
+import ch.epfl.bluebrain.nexus.kg.core.schemas.{SchemaId, SchemaName}
+import ch.epfl.bluebrain.nexus.kg.indexing.ElasticIndexingSettings
+import io.circe.Json
+
+/**
+  * Base class implementing common Elastic Search queries
+  *
+  * @param elasticClient  Elastic Search client
+  * @param settings       Elastic Search settings
+  * @param rs             HTTP client
+  * @tparam F             the monadic effect type
+  * @tparam Id            type of the ID that's returned
+  */
+abstract class BaseElasticQueries[F[_], Id](elasticClient: ElasticClient[F], settings: ElasticIndexingSettings)(
```

Comment:
Is there a reason why `fields` is not used to return certain fields? I've seen that you call elasticsearch listings only when fields are `Field.Empty`.
In case of Fields.Empty we want to retrieve only the `@id`, so it would be more performant to specify this on the calls to `client.search...`

Reply:
I wanted to implement the minimal feature set for this PR, and possibly add more features in following PRs.